backend/src/app/services/snapshot/snapshot.service.ts

```diff
@@ -151,6 +152,121 @@ export class SnapshotsService {
     }
   }
 
+  getNotatioParticipantsForSnapshotCreation = async (notationId: string) => {
+    try {
+      if (notationId === '' || notationId === null) {
+        throw Error('notation id cannot be empty');
+      }
+
+      return await this.eventsParticipantsRepo.find({
+        where: {
+          eventId: notationId,
+          srAction: SRApprovalStatusEnum.PUBLIC,
+        },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getNotationsForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return await this.eventsRepositoryRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getSiteParticipantsForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return await this.siteParticipantsRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getSiteDocumentsForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return await this.siteDocumentsRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getLandHisotoriesForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return this.landHistoriesRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getDisclosureForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return this.siteProfilesRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getSubDivisionsForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
+      }
+
+      return await this.siteSubDivisionsRepo.find({
+        where: { siteId, srAction: SRApprovalStatusEnum.PUBLIC },
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
+  getSiteAssociationsForSnapshotCreation = async (siteId: string) => {
+    try {
+      if (siteId === '' || siteId === null) {
+        throw Error('site id cannot be empty');
```

Comment:
Please implement HTTP exceptions with meaningful descriptive messages to enhance clarity and improve error handling.